### PR TITLE
12 Inferential Stats P01

### PR DESCRIPTION
**Pull request by JCardenasRdz**
Inferential Statistic is done

### CODE REVIEW

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        print()
+
+# ANOVA age vs pCR
+import seaborn as sns
+from scipy import stats
+
+def fig_01(ydata,xdata, hue_data):
+    sns.boxplot(x= xdata, y=ydata, hue = hue_data, data=df, palette="Set3");
+
+def anova_(predictor):
```
Comment (amirziai) [inline Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py:69]: Is `anova_` conflicting with another `anova` in the namespace?

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    print('ANOVA for ' + predictor + ' on PCR')
```
Comment (amirziai) [inline Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: Consider using string interpolation e.g. `print('ANOVA for {} on PCR'.format(predictor))`

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    f, p = stats.f_oneway(df.loc[df.PCR=='No',:][predictor],
```
Comment (amirziai) [inline Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: `, :]` vs `,:]`

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        T = pd.crosstab(df[var], 
+                        df[outcome]).astype(float);
+        chi2, p, dof, ex  = chi2_contingency( T.values )
+        odds_ratio, _  = fisher_exact(T.values.T)
+        print(T)
+        p_print(p, var, outcome,odds_ratio)
+        print()
+
+# ANOVA age vs pCR
+import seaborn as sns
```
Comment (amirziai) [inline Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: Move all imports to the beginning of the file

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+# load data (already cleaned)
+df = pd.read_csv('ISPY_clinical_clean.csv');
+df.dropna(axis=0, inplace=True)
+
+# func
+def heading(str_):
+    print("")
+    print(10*"-" + str_ + 10*"-")
+def footer():
```
Comment (amirziai) [inline Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py:16]: Add a line above

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import pandas as pd
+from scipy.stats import chi2_contingency
+from scipy.stats import fisher_exact
+import matplotlib as plt
+
+# load data (already cleaned)
+df = pd.read_csv('ISPY_clinical_clean.csv');
+df.dropna(axis=0, inplace=True)
```
Comment (amirziai) [inline Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: http://stackoverflow.com/questions/22532302/pandas-peculiar-performance-drop-for-inplace-rename-after-dropna

**Review by amirziai — APPROVED** [review]
Looks great. A few comments to consider.

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py (hunk 6 above)
Comment (JCardenasRdz) [reply Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: Got it.

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py (hunk 4 above)
Comment (JCardenasRdz) [reply Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: done

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py (hunk 1 above)
Comment (JCardenasRdz) [reply Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py:69]: I don't think so

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py (hunk 2 above)
Comment (JCardenasRdz) [reply Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: done

File: Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py (hunk 3 above)
Comment (JCardenasRdz) [reply Projects/12-Inferential_Stats_P01/Inferential_Stats_P01.py]: done

**Pull request merged**